pyccel/codegen/compiling/compilers.py

```diff
@@ -333,7 +333,12 @@ def compile_program(self, compile_obj, output_folder, verbose = False):
         # Get compile options
         exec_cmd, includes, libs_flags, libdirs_flags, m_code = \
                 self._get_compile_components(compile_obj, accelerators)
-        linker_libdirs_flags = ['-Wl,-rpath' if l == '-L' else l for l in libdirs_flags]
+        if self._info['exec'] in ('nvcc', 'nvc', 'nvfortran'):
+            linker_libdirs_flags = [f'"-Wl,-rpath,{l[2:]}"' for l in libdirs_flags]
+            linker_libdirs_flags = self._insert_prefix_to_list(linker_libdirs_flags, '-Xcompiler')
```

Comment:
I tried to use `-Xcompiler` flag but it didn't work what does the `l[2:]` do?

Reply:
It gets rid of the "-L". As there are multiple comments being passed Xcompiler is really picky about the format. It has to be
```
-Xcompiler "-Wpath,-rpath,/my/path"
```
The lack of spaces made me change how the `-L` was added. But this does not seem to be necessary now that I see the correct format. I will change for something closer to the original code

Reply:
Fixed [3bad499](https://github.com/pyccel/pyccel-cuda/pull/14/commits/3bad4994545684aaed544abec41aa141107406bf)